shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java

```diff
@@ -227,27 +241,120 @@
+        return result;
+    }
+
+    @Override
+    public ASTNode visitDesc(final DescContext ctx) {
+        TableSegment table = (TableSegment) visit(ctx.tableName());
+        DescribeStatement result = new DescribeStatement();
+        result.setTable(table);
+        return result;
+    }
+
+    @Override
+    public ASTNode visitShowDatabases(final ShowDatabasesContext ctx) {
+        ShowDatabasesStatement result = new ShowDatabasesStatement();
+        ShowLikeContext showLikeContext = ctx.showLike();
+        if (null != showLikeContext) {
+            ShowLikeSegment showLikeSegment = (ShowLikeSegment) visit(ctx.showLike());
+            result.getAllSQLSegments().add(showLikeSegment);
+        }
+        return result;
+    }
+
+    @Override
+    public ASTNode visitShowTables(final ShowTablesContext ctx) {
+        ShowTablesStatement result = new ShowTablesStatement();
+        FromSchemaContext fromSchemaContext = ctx.fromSchema();
+        ShowLikeContext showLikeContext = ctx.showLike();
+        if (null != fromSchemaContext) {
+            FromSchemaSegment fromSchemaSegment = (FromSchemaSegment) visit(ctx.fromSchema());
+            result.getAllSQLSegments().add(fromSchemaSegment);
+        }
+        if (null != showLikeContext) {
+            ShowLikeSegment showLikeSegment = (ShowLikeSegment) visit(ctx.showLike());
+            result.getAllSQLSegments().add(showLikeSegment);
+        }
+        return result;
     }
     
     @Override
     public ASTNode visitShowTableStatus(final ShowTableStatusContext ctx) {
-        ShowTableStatusStatement showTableStatusStatement = new ShowTableStatusStatement();
+        ShowTableStatusStatement result = new ShowTableStatusStatement();
         FromSchemaContext fromSchemaContext = ctx.fromSchema();
         ShowLikeContext showLikeContext = ctx.showLike();
         if (null != fromSchemaContext) {
             FromSchemaSegment fromSchemaSegment = (FromSchemaSegment) visit(ctx.fromSchema());
-            showTableStatusStatement.getAllSQLSegments().add(fromSchemaSegment);
+            result.getAllSQLSegments().add(fromSchemaSegment);
         }
         if (null != showLikeContext) {
             ShowLikeSegment showLikeSegment = (ShowLikeSegment) visit(ctx.showLike());
-            showTableStatusStatement.getAllSQLSegments().add(showLikeSegment);
+            result.getAllSQLSegments().add(showLikeSegment);
         }
-        return showTableStatusStatement;
+        return result;
     }
-    
+
+    @Override
+    public ASTNode visitShowColumns(final ShowColumnsContext ctx) {
+        ShowColumnsStatement result = new ShowColumnsStatement();
+        FromTableContext fromTableContext = ctx.fromTable();
+        FromSchemaContext fromSchemaContext = ctx.fromSchema();
+        ShowLikeContext showLikeContext = ctx.showLike();
+        if (null != fromTableContext) {
+            FromTableSegment fromTableSegment = (FromTableSegment) visit(fromTableContext);
```

Comment:
Since `FromTableSegment` is not handled in `rewriteEngine`, `Show columns from t_order` could not be rewriten to correct sql. You can have a try.

Reply:
Please check whether there is simiar issue here. Thx.